File: csharp/0045-jump-game-ii.cs
public class Solution {
    public int Jump(int[] nums) {
        int left = 0, right = 0, res = 0;
        
        while (right < nums.Length - 1) {
            int maxJump = 0;
            for (int i = left; i <= right; i++) {
                maxJump = Math.Max(maxJump, i + nums[i]);
            }
            left = right + 1;
            right = maxJump;
            res++;
        }
        return res;
    }
}

File: swift/0045-jump-game-ii.swift
class Solution {
    func jump(_ nums: [Int]) -> Int {
        var left = 0, right = 0
        var res = 0
        
        while right < nums.count - 1 {
            var maxJump = 0
            for i in left...right {
                maxJump = max(maxJump, i + nums[i])
            }
            left = right + 1
            right = maxJump
            res += 1
        }
        return res
    }
}